### ISSUE TITLE
libpng warning: iCCP: known incorrect sRGB profile - 3 days, 2 Windows reinstalls, 2 computers, 1 Hantek 6022BL, and a ton of lost hair.

### ISSUE THREAD

**Issue by rawrsy** [OpenHantek/openhantek#274]
Hi guys. This has been driving me insane. All the other oscilloscope software works Hantek software works. PulseView works. The simple one works. But I *want* to use OpenHantek. But I can't.

I know very little about software, and I have no business being on github, or building my own installers. But I have RTFM'd every manual I can, every guide I can. I have researched this problem and I still have no idea what is going on.

The driver is installed, the driver works. The software sees the scope. Then it spits the dummy, and I have to close it. Ran it as administrator, from command line, ran with --useGLES switch, nothing works. Nothing has worked. I have had no joy whatsoever, and worse, I have no idea what this problem means. I cannot find any PNGs to speak of. I have reinstalled Windows twice to try and get this thing to work. OpenHantek is the only program I have on this damn computer right now and it still does not work. I am at my wits end.

I know this is not a channel for support, it is a channel for reporting bugs, but can someone *please* throw me a bone? I am drowning here. I had extremely high hopes for this little scope, and I was so excited to buy it once I learned of the huge third party community behind it. _Finding this software is what made me purchase this scope to begin with,_ And it is driving me absolutely bananas. I am not spending another second diagnosing this problem, until I have something to work with.

I am using the latest build. I am using the latest versions of Windows, fully patched. I have disabled driver signing. I have done everything I can and nothing has worked. Here is a log of what happens.

`PS C:\Users\wolf\Documents\Hantek Stuff\OpenHantek-x64-Release-b33 (3)> .\OpenHantek.exe --useGLES

stop CPU

... WROTE: 792 bytes, 10 segments, avg 79

reset CPU

2nd stage: write external memory

stop CPU

2nd stage: write on-chip memory

... WROTE: 16384 bytes, 17 segments, avg 963
reset CPU

libpng warning: iCCP: known incorrect sRGB profile

PS C:\Users\wolf\Documents\Hantek Stuff\OpenHantek-x64-Release-b33 (3)>`

Please, give me a clue. Something. Anything. I am desperate.

Cheers.
Rory.

**Comment by davidgraeff** [comment]
The software was written for Linux and MacOS and just happen to work on Windows as well. But there is a longer discussion in one of the windows issues on how to get it to work. If you don't see the device listed in the initial screen, your driver installation is not correct.

**Comment by rawrsy** [comment]
Hi David, here is a screenshot. I hope this helps illustrate the issue I'm having. Thanks for the amazingly quick reply!

It does see the scope initially, and then it wigs out and says it failed to connect. It *does* see it, but I can't capture the screenshot fast enough. It tries to upload the firmware and then runs into its problem.

Also, I would like to try this scope on the Mac. But what would I be looking at to get it work on that? I will RTFM and give it a go, though. But I would really love to make it work on the Windows laptop, that would be mega handy.
![openhantek problem](https://user-images.githubusercontent.com/32505439/47918347-0d3a9980-def8-11e8-9d7c-b1bc6a022f9a.png)

Thank you again!
Rory

**Comment by bikermikem** [comment]
You might want to look at http://www.hantek.com/en/PagesFW_Vyyrj.html

**Comment by rawrsy** [comment]
Yes, in having a play around, it does seem that the Hanteks are a bit.. weird. Am I right in suggesting that when these things boot, the driver spits the firmware at the device every time? I mean the firmware is all of 15KB so it's not exactly a bad way to go, and it will explain the zombie modes when I was fighting with the factory drivers. You'd pull the device up in either Hantek's software, or PulseView, it didn't matter, and it would give you *perfect* phantom square waves that didn't exist. 1MHz on the dot, zero rise/fall time, that sort of thing. A bit suspicious, to say the least...

So, do you think that my problem may still lie with the driver? And the program will still run with that libpng error? Because if so, then I don't care about libpng. If the program still works, that's a rabbit hole I'm not going to chase down, because I don't care. But if you think there are still driver issues, I might have a further play with it.

Cheers,
Rory

**Comment by Ho-Ro** [comment]
The libpng warning comes from this picture: `openhantek/res/images/switch.png`
There is a patch in my repo:
https://github.com/OpenHantek/OpenHantek6022/commit/f7744eb0690f338ad11cc70ae9aa4de166728b61

Martin
